app/models/manageiq/providers/google/refresh_helper_methods.rb

```diff
@@ -2,15 +2,20 @@ module ManageIQ::Providers::Google::RefreshHelperMethods
   extend ActiveSupport::Concern
 
   def process_collection(collection, key)
+    process_flatten_collection(collection, key) { |x| [yield(x)] }
+  end
+
+  def process_flatten_collection(collection, key)
```

Comment:
hm, might be nicer to just use one process_collection method

then you can extract lines 9 to 13 to process_item

and then 
parsed_item = yield(item)
parsed_item.kind_of?(Array) ? parsed_item.each {|x| process_item(x)} : process_item(parsed_item)